packages/google_maps_flutter/lib/src/marker.dart

```diff
@@ -55,33 +33,135 @@ class InfoWindowText {
   /// A null value means no additional text.
   final String snippet;
 
-  dynamic _toJson() => <dynamic>[title, snippet];
+  /// The icon image point that will be the anchor of the info window when
+  /// displayed.
+  ///
+  /// The image point is specified in normalized coordinates: An anchor of
+  /// (0.0, 0.0) means the top left corner of the image. An anchor
+  /// of (1.0, 1.0) means the bottom right corner of the image.
+  final Offset anchor;
+
+  /// onTap callback for this [InfoWindow].
+  final VoidCallback onTap;
+
+  /// Creates a new [InfoWindow] object whose values are the same as this instance,
+  /// unless overwritten by the specified parameters.
+  InfoWindow copyWith({
+    String titleParam,
+    String snippetParam,
+    Offset anchorParam,
+    VoidCallback onTapParam,
+  }) {
+    return InfoWindow(
+      title: titleParam ?? title,
+      snippet: snippetParam ?? snippet,
+      anchor: anchorParam ?? anchor,
+      onTap: onTapParam ?? onTap,
+    );
+  }
+
+  dynamic _toJson() {
+    final Map<String, dynamic> json = <String, dynamic>{};
+
+    void addIfPresent(String fieldName, dynamic value) {
+      if (value != null) {
+        json[fieldName] = value;
+      }
+    }
+
+    addIfPresent('title', title);
+    addIfPresent('snippet', snippet);
+    addIfPresent('anchor', _offsetToJson(anchor));
+
+    return json;
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is InfoWindow &&
+          runtimeType == other.runtimeType &&
+          title == other.title &&
+          snippet == other.snippet &&
+          anchor == other.anchor;
+
+  @override
+  int get hashCode => title.hashCode ^ snippet.hashCode ^ anchor.hashCode;
+
+  @override
+  String toString() {
+    return 'InfoWindow{title: $title, snippet: $snippet, anchor: $anchor}';
+  }
 }
 
-/// Configuration options for [Marker] instances.
+/// Uniquely identifies a [Marker] among [GoogleMap] markers.
 ///
-/// When used to change configuration, null values will be interpreted as
-/// "do not change this configuration option".
-class MarkerOptions {
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class MarkerId {
+  MarkerId(this.value) : assert(value != null);
+
+  /// value of the [MarkerId].
+  final String value;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is MarkerId &&
+          runtimeType == other.runtimeType &&
+          value == other.value;
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  String toString() {
+    return 'MarkerId{value: $value}';
+  }
+}
+
+/// An icon placed at a particular geographical location on the map's surface.
```

Comment:
Is a marker "an icon"?
Maybe just say in the summary "Marks a geographical location on the map"?

[I see that you just moved this, but might be a good opportunity anyway]